app/services/github/contributions/pull_request_collector.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Github
+  module Contributions
+    class PullRequestCollector
+      LINK_INDEX = {
+        org: 3,
+        repo: 4,
+        pr: 6
+      }
+
+      def initialize(client:)
+        @client = client
+      end
+
+      def all
+        valid_pull_requests = Contribution.valid_pull_requests.select(:id, :link, :pr_state)
+        updated_pull_request = valid_pull_requests.map {|contribution| { id: contribution.id, pr_state: new_pr_state(contribution.link)} }
+
+        updated_pull_request.select{ |new_pr| validate_update(valid_pull_requests.find_by(id: new_pr[:id]), new_pr) }
+      end
+
+      private
+
+      attr_reader :client
+
+      def validate_update(current_pr, new_pr)
+        return false if new_pr[:pr_state] == 'Fetch Error'
```

Comment:
Oh yeah, something I forgot to mention on the previous review: the strings used to determine the pr_state should be added as constants as well since you'll be using them in more than one place.

For instance `'Fetch Error'` is being called in here and on line 40 as well, it would be ideal that if we ever change the name of this pr_state we need to change it only one time.

Reply:
Ok!